TESTING.md

```diff
@@ -90,6 +139,29 @@ This entity is initialized when the test starts. You can assume that the variabl
 The `check` variable is our interface with the `check.v1` package. We can do several things with it, such as skipping a test, probing a condition, declare success or failure, and more.
 
 
+### Adding build tags.
+
+All tests need to have a build tag. The tag should be the first line of the file, followed by a blank line
+
+```go
+// +build functional featurename ALL
+
+package govcd
+```
+
+Tests that integrate in the existing suite use the tag `functional`. Using that tag, we can run all functional tests at once.
```

Comment:
The paragraph about this "existing suite" is confusing. If I'm a newcomer what is "existing"? It sort of implies there's some "non-existing" tests that one doesn't know of. It would be much more useful to describe, that functional tests are the ones that require vCD connection, while unit tests - don't. In other words, describe the types of tests as opposed to history.

I suggest we have three bulleted paragraphs here:
- That all test need to have either `functional` or `unit` (currently, the confusing/missing part)
- All should have `ALL` (already described)
- About the "feature" tags (already described)

Reply:
I have rephrased the paragraph to describe the scope and usage of the framework.